api/pkg/setup/setup.go

```diff
@@ -162,12 +184,26 @@ func GetConfiguration() (*Configuration, error) {
 		}
 	}
 
-	cfg := &Configuration{}
-	err = vpr.Unmarshal(cfg)
+	// unique case where we want to be able to configure static OIDC providers
```

Comment:
@hspitzley-czi I got this working, although is the behavior you'd expect? Or should I just allow the environment variable HAPI_OIDCPROVIDERS override whatever is the configuration file? I feel like having a magic environment variable the overwrites stuff seems odds, but I'm not sure. I stared at viper too long today

Reply:
If you can think of a more general purpose way of doing this rather than only for the auth providers, that might be helpful down the line. Like something we could also use for other fields in this config file.

Reply:
🤔 I think I am going to keep this for now, but something doesn't feel correct to me. Just haven't identified it yet